src/transformers/models/llama/modeling_llama.py

```diff
@@ -692,27 +754,18 @@ def forward(
 
         # Is this the input to the model?
         hidden_states = inputs_embeds
-        if self.spmd_2d_sharding > 0:
-            # Apply 2D sharding:
-            # input (data,, None, model)
-            import torch_xla.core.xla_model as xm
-            import torch_xla.experimental.xla_sharding as xs
-            import torch_xla.runtime as xr
-            import torch_xla
-            num_devices = xr.global_runtime_device_count()
-            device_ids = torch.arange(num_devices)
-            if self.spmd_debug:
-                print('> Sharding hidden_states', hidden_states.shape)
-            model = self.spmd_2d_sharding
-            data = num_devices // model
-            assert model * data == num_devices
-            if self.spmd_iota_mesh:
-                mesh = xs.Mesh(device_ids, (data, 1, model))
-            else:
-                mesh = xs.HybridMesh(ici_mesh_shape=(data, 1, model))
-            xs.mark_sharding(hidden_states, mesh, (0, 1, 2))
-            if self.spmd_debug:
-                print(torch_xla._XLAC._get_xla_sharding_spec(hidden_states))
+        # Apply 2D sharding:
+        # hidden_states (batch, length, hidden)
+        # mesh (data, None, model)
+        if self.spmd_iota_mesh:
+            data_model_mesh = xs.Mesh(device_ids, (self.spmd_data_axis, 1, self.spmd_model_axis))
```

Comment:
`NameError: name 'device_ids' is not defined`

Reply:
Good catch. I'm trying to integrate spmd_iota_mesh now.